intake_xarray/netcdf.py

```diff
@@ -1,33 +1,39 @@
 # -*- coding: utf-8 -*-
 import xarray as xr
-from .base import DataSourceMixin
+from .base import XarraySource
 
 
-class NetCDFSource(DataSourceMixin):
-    """Open a xarray file.
+class NetCDFSource(XarraySource):
+    """Open a netcdf file.
+
+    This creates an xarray.Dataset.
+
+    See http://xarray.pydata.org/en/stable/generated/xarray.open_dataset.html
+    for the file formats supported and possible extra arguments.
+
+    NOTE: When reading from OpenDAP URLs do not set the ``chunks`` option to
+    use provided default chunking.
 
     Parameters
     ----------
-    urlpath: str
-        Path to source file. May include glob "*" characters. Must be a
-        location in the local file-system.
-    chunks: int or dict
-        Chunks is used to load the new dataset into dask
-        arrays. ``chunks={}`` loads the dataset with dask using a single
-        chunk for all arrays.
+    urlpath : str
+        Path to source file. May include glob "*" characters or format
+        pattern strings. Accepts un-authenticated OpenDAP urls, and any
+        remote source can be used as long as caching is enabled.
+        Must be a format supported by ``xr.open_dataset``.
+
+        Some examples:
+            - ``s3://data/*.nc``
+            - ``http://thredds.ucar.edu/thredds/dodsC/grib/FNMOC/WW3/Global_1p0deg/Best``
+            - ``https://github.com/pydata/xarray-data/blob/master/air_temperature.nc?raw=true``
```

Comment:
So does this handle remote URLs directly or not? I assume if it is opendap, then yes, in which case the thing about needing caching is wrong (and in fact won't work).
The thredds URL gives Unrecognized Request for me.